### PR TITLE
Build wheel on Mac

### PR DESCRIPTION
**Pull request by wenleix**
Uploaded as https://pypi.org/project/torcharrow/0.0.2.dev20211122/#files

Can be installed with 
```python
pip install torcharrow==0.0.2.dev20211122 --user
```

### CODE REVIEW

File: packaging/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+### Linux
+Coming soon!
+
+### OSX
+Build the wheel:
+```
+PYTHON_VERSION=3.8 packaging/build_wheel.sh
+```
+
+Fix the wheel with `delocate`:
```
Comment (wenleix) [inline packaging/README.md:17]: Here are the dynamic libraries being included/relocated:
```
(torcharrow) wxie@wxie-mbp torcharrow %  delocate-listdeps dist/torcharrow-0.0.2.dev20211122-cp38-cp38-macosx_10_15_x86_64.whl
/usr/local/Cellar/boost/1.76.0/lib/libboost_context-mt.dylib
/usr/local/Cellar/boost/1.76.0/lib/libboost_filesystem-mt.dylib
/usr/local/Cellar/boost/1.76.0/lib/libboost_program_options-mt.dylib
/usr/local/Cellar/boost/1.76.0/lib/libboost_regex-mt.dylib
/usr/local/Cellar/boost/1.76.0/lib/libboost_system-mt.dylib
/usr/local/Cellar/boost/1.76.0/lib/libboost_thread-mt.dylib
/usr/local/Cellar/gflags/2.2.2/lib/libgflags.2.2.2.dylib
/usr/local/Cellar/glog/0.5.0/lib/libglog.0.5.0.dylib
/usr/local/Cellar/icu4c/69.1/lib/libicudata.69.1.dylib
/usr/local/Cellar/icu4c/69.1/lib/libicui18n.69.1.dylib
/usr/local/Cellar/icu4c/69.1/lib/libicuuc.69.1.dylib
/usr/local/Cellar/libevent/2.1.12/lib/libevent-2.1.7.dylib
/usr/local/Cellar/lz4/1.9.3/lib/liblz4.1.9.3.dylib
/usr/local/Cellar/openssl@1.1/1.1.1l_1/lib/libcrypto.1.1.dylib
/usr/local/Cellar/openssl@1.1/1.1.1l_1/lib/libssl.1.1.dylib
/usr/local/Cellar/protobuf/3.17.3/lib/libprotobuf.28.dylib
/usr/local/Cellar/snappy/1.1.9/lib/libsnappy.1.1.9.dylib
/usr/local/Cellar/zstd/1.5.0/lib/libzstd.1.5.0.dylib
```

Would be great if Velox could allow dependency library to statically linked with these libraries. cc @pedroerp , @damianr99

File: packaging/build_wheel.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    export CC=clang CXX=clang++
+  fi
+}
+
+# Inputs:
+#   PYTHON_VERSION (3.8, 3.9)
+#
+# Outputs:
+#   PATH modified to put correct Python version in PATH
+setup_wheel_python() {
```
Comment (ejguan) [inline packaging/build_wheel.sh:41]: Wait, is this line being executed?

Comment (wenleix) [reply packaging/build_wheel.sh:41]: @ejguan : good catch. fixed ^_^

**Review by ejguan — APPROVED** [review]
It generally looks good to me.
You may also need to support wheels based on the matrix of macos versions and python versions in the future.

**Pull request merged**